frontend/routes/viewer/components/gis/gis.component.tsx

```diff
@@ -0,0 +1,289 @@
+		if (!this.state.settingsModeActive) {
+			return [ { Button: this.getMenuButton } ];
+		}
+		return [];
+	}
+
+	public renderSettings = () => {
+		let formValues = {} as any;
+		const settings = this.props.settings;
+
+		if (settings.surveyPoints && settings.surveyPoints.length) {
+			const [{
+				position: [axisX, axisY, axisZ],
+				latLong: [latitude, longitude]
+			}] = settings.surveyPoints;
+
+			formValues = { axisX, axisY, axisZ, latitude, longitude };
+		}
+
+		formValues.angleFromNorth = settings.angleFromNorth || 0;
+
+		const EnhancedSettingsForm = withFormik({
+			mapPropsToValues: () => (formValues),
+			handleSubmit: (values) => {
+				const { angleFromNorth, axisX, axisY, axisZ, latitude, longitude } = values;
+
+				const pointsSettings = {
+					angleFromNorth,
+					surveyPoints: [{
+						position: [axisX, axisY, axisZ],
+						latLong: [latitude, longitude]
+					}]
+				};
+
+				const { teamspace, modelId } = this.getDataFromPathname();
+				const project = localStorage.getItem('lastProject');
+				const modelData = { teamspace, project, modelId };
+
+				this.props.updateModelSettings(modelData, pointsSettings);
+			},
+			validationSchema: SettingsSchema
+		})(SettingsForm);
+
+		return <EnhancedSettingsForm />;
+	}
+
+	public handleChangeMapProvider = (event) => {
+		this.setState({
+			activeMapIndex: event.target.value
+		}, () => {
+			this.props.resetSources();
+		});
+	}
+
+	public renderVisibilityButton = (layer) => {
+		return (
+			<VisibilityButton>
+				{includes(this.state.visiblieSources, layer.source)
+					? <VisibilityIcon onClick={() => this.props.removeSource(layer.source)}/>
+					: <VisibilityOffIcon onClick={() => this.props.addSource(layer.source)} />
+				}
+			</VisibilityButton>
+		);
+	}
+
+	public renderMapProviders = (mapsProviders = []) =>
+		mapsProviders.map((mapProvider, index) => (
+			<MenuItem key={mapProvider.name} value={index}>
+				{mapProvider.name}
+			</MenuItem>
+		))
+
+	public renderLayers = (mapLayers = []) =>
+		mapLayers.map((layer) => (
+			<MapLayer key={layer.name}>
+				<MapNameWrapper>
+					<StyledMapIcon color="inherit" />
+					<MapName>{layer.name}</MapName>
+				</MapNameWrapper>
+
+				{this.renderVisibilityButton(layer)}
+			</MapLayer>
+		))
+
+	public renderMapLayers = () => {
+		const { mapsProviders } = this.props;
+		const { activeMapIndex } = this.state;
+
+		return (
+			<ViewerPanelContent>
+				<StyledSelect
+					onChange={this.handleChangeMapProvider}
+					value={activeMapIndex}>
+					{this.renderMapProviders(mapsProviders)}
+				</StyledSelect>
+				{this.renderLayers(mapsProviders[activeMapIndex].layers)}
+			</ViewerPanelContent>
+		);
+	}
+
+	public render() {
+		const { settingsModeActive } = this.state;
+
+		return (
+			<ViewerPanel
+				title="GIS"
+				Icon={this.getTitleIcon()}
+				actions={this.getActions()}
+				pending={this.props.isPending}
+			>
+				{settingsModeActive ? this.renderSettings() : this.renderMapLayers()}
```

Comment:
I'd separate renderSettings() and renderMapLayers() into to differents components/files.

Reply:
I moved Settings to separate component, but referring to the layers, we think that it's so small that we don't need to divide it into smaller ones